src/suites/cts/validation/createView.spec.ts

```diff
@@ -0,0 +1,268 @@
+      dimension: '2d',
+      format: FORMAT,
+      usage: GPUTextureUsage.SAMPLED,
+    });
+  }
+
+  getDescriptor(
+    options: {
+      format?: GPUTextureFormat;
+      dimension?: GPUTextureViewDimension;
+      baseMipLevel?: number;
+      mipLevelCount?: number;
+      baseArrayLayer?: number;
+      arrayLayerCount?: number;
+    } = {}
+  ): GPUTextureViewDescriptor {
+    const {
+      format = FORMAT,
+      dimension = '2d',
+      baseMipLevel = 0,
+      mipLevelCount = MIP_LEVEL_COUNT,
+      baseArrayLayer = 0,
+      arrayLayerCount = 1,
+    } = options;
+    return {
+      format,
+      dimension,
+      baseMipLevel,
+      mipLevelCount,
+      baseArrayLayer,
+      arrayLayerCount,
+    };
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('creating texture view on a 2D non array texture', async t => {
+  const { dimension = '2d', arrayLayerCount, mipLevelCount, baseMipLevel, success } = t.params;
+
+  const texture = t.createTexture({ arrayLayerCount: 1 });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+    mipLevelCount,
+    baseMipLevel,
+  });
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { arrayLayerCount: 1, success: true }, // it is OK to create a 2D texture view on a 2D texture
+  { arrayLayerCount: 2, success: false }, // it is an error to view a layer past the end of the texture
+  { dimension: '2d-array', arrayLayerCount: 1, success: true }, // it is OK to create a 1-layer 2D array texture view on a 2D texture
+  // baseMipLevel == k && mipLevelCount == 0 means to use levels k..end
+  { mipLevelCount: 0, baseMipLevel: 0, success: true },
+  { mipLevelCount: 0, baseMipLevel: 1, success: true },
+  { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT - 1, success: true },
+  { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT, success: false },
+  // it is an error to make the mip level out of range
+  { mipLevelCount: MIP_LEVEL_COUNT + 1, baseMipLevel: 0, success: false },
+  { mipLevelCount: MIP_LEVEL_COUNT, baseMipLevel: 1, success: false },
+  { mipLevelCount: 2, baseMipLevel: MIP_LEVEL_COUNT - 1, success: false },
+  { mipLevelCount: 1, baseMipLevel: MIP_LEVEL_COUNT, success: false },
+]);
+
+g.test('creating texture view on a 2D array texture', async t => {
+  const { dimension = '2d-array', arrayLayerCount, baseArrayLayer, success } = t.params;
+
+  const texture = t.createTexture({ arrayLayerCount: ARRAY_LAYER_COUNT_2D });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+    baseArrayLayer,
+  });
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { dimension: '2d', arrayLayerCount: 1, success: true }, // it is OK to create a 2D texture view on a 2D array texture
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D, success: true }, // it is OK to create a 2D array texture view on a 2D array texture
```

Comment:
This should be failing because there is an explicit `arrayLayerCount`, so the user has to specify the `dimension` (or they get `2d` by default).

Reply:
`getDescriptor` actually set dimension to '2d'. This is why it is succeeding here.